repoclosure/repoclosure.sh

```diff
@@ -3,35 +3,43 @@
 set -e
 
 if [ $# -lt 2 ]; then
-  echo "Usage: $0 <yum.conf> <url>"
-  exit 1
+	echo "Usage: $0 <dnf.conf> <releasever>"
+	exit 1
 fi
 
-yumorig=$1
-url=$2
-shift; shift
+config=$1 ; shift
+releasever=$1 ; shift
 
-TEMPDIR=$(mktemp -d)
+TEMPDIR=$(mktemp -d -t repoclosure-XXXXXX)
 trap "rm -rf $TEMPDIR" EXIT
 
-# repo names must be unique, or yum will get confused between different OSes and URLs
-reponame=undertest-$(basename $yumorig .conf)-$(echo $url | cksum | sed 's/ /-/g')
-
-yumconf=$TEMPDIR/yum.conf
-cat $yumorig > $yumconf
-cat >> $yumconf << EOF
-
-[$reponame]
-name=$reponame
-gpgcheck=0
-baseurl=$url
-
-EOF
-
-repoclosure -c $yumconf -t -r $reponame $* 2>&1 | tee $TEMPDIR/repoclosure.log
-
-if tail -n1 $TEMPDIR/repoclosure.log | grep -q "Num Packages"; then
-  exit 0
-else
-  exit 1
+if [[ $releasever == 8 ]] ; then
+	# Enable modularity on EL8
+	MODULESDIR=$TEMPDIR/etc/dnf/modules.d
+	mkdir -p $MODULESDIR
+	cat > $MODULESDIR/ruby.module <<-EOF
+	[ruby]
+	name=ruby
+	stream=2.7
+	profiles=
+	state=enabled
+	EOF
+
+	cat > $MODULESDIR/postgresql.module <<-EOF
+	[postgresql]
+	name=postgresql
+	stream=12
+	profiles=
+	state=enabled
+	EOF
+
+	cat > $MODULESDIR/foreman.module <<-EOF
+	[foreman]
+	name=foreman
+	stream=el$releasever
+	profiles=
+	state=enabled
+	EOF
```

Comment:
I take it from this that if the foreman module was the only one in the config, it would not cause its dependencies to get enabled and thus we must hard-code the ruby and postgresql module configs in?

Reply:
Correct - this is also what `dnf module enable foreman:el8` does. If you don't, you get conflicts. I've tried, but sadly we'll have this duplication.

I'm not sure yet how we should deal with Katello, which we don't want to enable unless it should.

Reply:
It may be possible to run `dnf --installroot $TEMPDIR --releasever $releasever module enable foreman` instead.

Reply:
It isn't possible:
```console
$ dnf --installroot /tmp/repoclosure-RSRkCQ --releasever 8 module enable -y foreman
Error: This command has to be run with superuser privileges (under the root user on most systems).
```